src/controller/CHIPDeviceController.cpp

```diff
@@ -1361,6 +1472,27 @@ CHIP_ERROR DeviceCommissioner::OnOperationalCredentialsProvisioningCompletion(De
     return CHIP_NO_ERROR;
 }
 
+void DeviceCommissioner::OnHandleEchoResponse(Messaging::ExchangeContext * ec, System::PacketBufferHandle && payload)
+{
+    uint64_t respTime    = static_cast<uint64_t>(System::Timer::GetCurrentEpoch());
+    uint64_t transitTime = respTime - mPingParams.GetLastEchoTime();
+
+    mPingParams.SetWaitingForEchoResp(false);
+    mPingParams.IncrementEchoRespCount();
+
+    ChipLogProgress(Controller, "Echo Response: %" PRIu32 "/%" PRIu32 "(%.2f%%) len=%u time=%.3fms\n",
```

Comment:
This does not scale to embedded to use float prints. Is there a way to avoid this showing-up on embedded devices once some have controller ability (e.g. to affect bindings) ?

Reply:
Do we have plan to enable controller on embedded platform?

Reply:
A lot of the patterns in the controller object (e.g. statically allocated memory for controlled devices) imply we are trying to accommodate embedded platform controllers.  RTOS-based Thread border routers seem a likely candidate.  More generally, the device-to-device model would require this be supportable.